File: _assets/scripts/jig.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "strings"
)

func main() {
    if len(os.Args) < 2 {
        fmt.Println("Usage:", os.Args[0], "<input_file>")
        os.Exit(1)
    }

    inputFile, err := os.Open(os.Args[1])
    if err != nil {
        fmt.Println("Error opening input file:", err)
        os.Exit(1)
    }
    defer inputFile.Close()

    scanner := bufio.NewScanner(inputFile)
    var pollster, month, state string
    for scanner.Scan() {
        line := scanner.Text()
        parts := strings.Split(line, "]")
        if len(parts) >= 3 && strings.HasPrefix(parts[0], "master_polls.data[") {
            pollsterAndMonth := strings.TrimPrefix(parts[0], "master_polls.data[")
            pollsterAndMonthParts := strings.Split(pollsterAndMonth, "][")
            if len(pollsterAndMonthParts) >= 2 {
                pollster = pollsterAndMonthParts[0]
                month = pollsterAndMonthParts[1]
                state = strings.TrimPrefix(parts[1], ".data[")
            } else {
                fmt.Println("Invalid pollster and month format, skipping line")
                continue
            }
        } else if strings.HasPrefix(line, "PollData(") {
            pollData := strings.TrimPrefix(line, "PollData(")
            pollData = strings.TrimSuffix(pollData, ")")
            pollValues := strings.Split(pollData, ",")
            if len(pollValues) == 3 {
                fmt.Printf("add_poll!(%s, %s, %s, %s, %s, %s)\n", month, state, pollster, pollValues[0], pollValues[1], pollValues[2])
            } else {
                fmt.Println("Invalid poll data format, skipping line")
            }
        }
    }

    if err := scanner.Err(); err != nil {
        fmt.Println("Error reading input file:", err)
        os.Exit(1)
    }
}

File: _assets/scripts/jig.rs
use std::env;
use std::fs::File;
use std::io::{BufRead, BufReader};

fn main() {
    let args: Vec<String> = env::args().collect();

    if args.len() < 2 {
        eprintln!("Usage: {} <input_file>", args[0]);
        std::process::exit(1);
    }

    let input_file = &args[1];
    let file = File::open(input_file).expect("Failed to open input file");
    let reader = BufReader::new(file);

    for line in reader.lines() {
        let line = line.expect("Failed to read line");
        let parts: Vec<&str> = line.split(&['[', ']', '=', ' ', '(', ')'][..]).filter(|s| !s.is_empty()).collect();

        if parts.len() >= 8 && parts[0] == "master_polls.data" {
            let pollster = parts[1];
            let month = parts[2];
            let state = parts[4];
            let poll_data: Vec<&str> = parts[6].split(',').collect();

            if poll_data.len() == 3 {
                let biden_support = poll_data[0];
                let trump_support = poll_data[1];
                let sample_size = poll_data[2];

                println!(
                    "add_poll!({}, {}, {}, {}, {}, {})",
                    month, state, pollster, biden_support, trump_support, sample_size
                );
            }
        }
    }
}